### PR TITLE
Run CI against Ruby 3.2

### PR DESCRIPTION
**Pull request by koheisg**

### CODE REVIEW

File: spec/lib/http/options/headers_spec.rb [hunk 1]
```diff
@@ -14,7 +14,11 @@
   end
 
   it "accepts any object that respond to :to_hash" do
-    x = Struct.new(:to_hash).new("accept" => "json")
+    x = if RUBY_VERSION >= "3.2.0"
+          Data.define(:to_hash).new(:to_hash => { "accept" => "json" })
+        else
+          Struct.new(:to_hash).new({ "accept" => "json" })
+        end
```
Comment (koheisg) [inline spec/lib/http/options/headers_spec.rb:21]: `Struct.new(:to_hash, keyword_init: false).new("accept" => "json")`

If this is the case, the code will not branch by version, but keyword_init: false will cause rubocop to say `Use hash rockets syntax.

Comment (koheisg) [reply spec/lib/http/options/headers_spec.rb:21]:
```suggestion
    Struct.new(:to_hash, keyword_init: false).new("accept" => "json")
```

`Struct.new(:to_hash, keyword_init: false).new("accept" => "json")`

If this is the case, the code will not branch by version, but keyword_init: false will cause rubocop to say `Use hash rockets syntax.

**Comment by tarcieri** [comment]
@koheisg a separate issue, but it's probably about time we changed from hashrockets to keyword args in general

**Comment by 1mmt-kohei-sugi** [comment]
@tarcieri OK ruby 3.2 support first! Thanks.

**Comment by koheisg** [comment]
@tarcieri OK ruby 3.2 support first! Thanks.

File: spec/lib/http/options/headers_spec.rb (hunk 1 above)
Comment (tarcieri) [reply spec/lib/http/options/headers_spec.rb:21]: I think we should migrate from hashrockets to keyword args in general, and that's long overdue.

I would suggest disabling the rubocop lint for hashrockets.

**Pull request merged**